base/dict.jl

```diff
@@ -191,9 +191,14 @@ end
 
 
 """
-    similar(d::Dict)
-
-Create an empty dictionary with the same key and value types as `d`.
+    similar(d::Dict, [element_type=eltype(e::Dict)])
+
+Create an empty dictionary with the same key and value types as `e`. The second element
+is a pair `Pair{K,V}` of key and value types and it is optional, defaulting to
+`eltype(d)`.
+Note that `similar` is useful when writing generic code that works with different
+`Associative` types. Instead, an empty dictionary with the element type of `e` is simply
+created with `Dict{keytype{e},valtype{e}}()`.
```

Comment:
Anyway I don't think this paragraph is useful, as it's much shorter to write `similar(d)`, so why discourage people from using it?